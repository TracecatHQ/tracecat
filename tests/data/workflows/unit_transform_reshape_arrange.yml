title: Reshape data
description: Test reshape data
entrypoint:
  ref: flatten
inputs:
  details:
    name: John Doe
    age: 30
    job:
      title: Software Engineer
      company: Acme Inc.
      city: Sometown
    residence:
      street_address: 123 Main St.
      city: Anytown
    contact:
      email: john@acme.com
      phone: 123-456-7890

actions:
  - ref: flatten
    action: core.transform.reshape
    # Flatten the mapping into a list of key-value pairs
    args:
      value:
        name: ${{ INPUTS.details.name }}
        age: ${{ INPUTS.details.age }}
        job_title: ${{ INPUTS.details.job.title }}
        job_company: ${{ INPUTS.details.job.company }}
        job_city: ${{ INPUTS.details.job.city }}
        residence_street_address: ${{ INPUTS.details.residence.street_address }}
        residence_city: ${{ INPUTS.details.residence.city }}
        contact_email: ${{ INPUTS.details.contact.email }}
        contact_phone: ${{ INPUTS.details.contact.phone }}
