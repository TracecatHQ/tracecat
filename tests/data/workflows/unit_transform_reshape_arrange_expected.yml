ACTIONS:
  flatten:
    result:
      name: John Doe
      age: 30
      job_title: Software Engineer
      job_company: Acme Inc.
      job_city: Sometown
      residence_street_address: 123 Main St.
      residence_city: Anytown
      contact_email: john@acme.com
      contact_phone: 123-456-7890
    result_typename: "dict"

INPUTS:
  details:
    name: John Doe
    age: 30
    job:
      title: Software Engineer
      company: Acme Inc.
      city: Sometown
    residence:
      street_address: 123 Main St.
      city: Anytown
    contact:
      email: john@acme.com
      phone: 123-456-7890
TRIGGER:
